### PR TITLE
Phase1 Pixel DQM Online, Raw Data, and Bugfix

### PR DESCRIPTION
**Pull request by schneiml**
As the title says, this PR adds 3 things to the Phase1 DQM:
- The pixel online configuration is modified to run the relevant Phase1 plugins as well. Doing this on the live config should be fine, since 8_1_0 should never go into online production, and for 2017 the new plugins are needed (and the old ones should be removed).
- Along with this come improvements to make the Phase1 DQM actually work in online. This includes harvesting running in online and configuring the Phase1 DQM to run on Phase0 data (there is no Phase1 online yet... -- the changes only affect online). 
- The Phase1-on-Phase0 config allows testing and using the Phase1 Raw Data monitoring, so I added it here. This is really only designed and tested for Phase0, since we don't have enough information about Phase1 yet. It can already run in Phase1 offline (the records are present) but obviously there is nothing to monitor in MC.
- There is a fairly critical bugfix for the Phase1 DQM framework. A c&p mistake in the last PR made all 2D profiles actually TH2s which is incorrect.

Still to be done are specific  plots and settings for online. There might also be a memory leak in the Phase1 harvesting code which would become a problem in online, but this is hard to tell at the moment (involves `new`ed TH1s and `TH1::add`).

### CODE REVIEW

**Comment by dmitrijus** [comment]
+1

File: DQM/SiPixelPhase1RawData/src/SiPixelPhase1RawData.cc [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+      int fed = siPixelRawDataError.getFedId();
+      int type = siPixelRawDataError.getType();
+      DetId id = it->detId();
+      
+      const uint32_t LINK_bits = 6;
+      const uint32_t LINK_shift = 26;
+      const uint64_t LINK_mask = (1 << LINK_bits) - 1;
+
+      uint64_t errorWord = 0;
+      if (type == 32 || type == 33 || type == 34) {
```
Comment (schneiml) [inline DQM/SiPixelPhase1RawData/src/SiPixelPhase1RawData.cc]: @davidlange6 I can't, I'm afraid. I just took the minmal subset needed to get the commonly used functionality out of https://github.com/cms-sw/cmssw/blob/CMSSW_8_1_X/DQM/SiPixelMonitorRawData/src/SiPixelRawDataErrorModule.cc . And that plugin is in a very sad shape, with lots more of these and a number of obvious mistakes (like the one fixed recently in 01341de52c2ee91e86e36).

For "real" phase1 we need to find some reliable documentation on the FED output and implement this properly.

**Comment by davidlange6** [comment]
maybe someone from the pixel community can help? otherwise, this might be relevant

https://github.com/cms-sw/cmssw/blob/CMSSW_8_1_X/DataFormats/SiPixelRawData/src/SiPixelRawDataError.cc

**Comment by boudoul** [comment]
Adding @dkotlins  @veszpv

**Comment by schneiml** [comment]
@davidlange6 can you live with the magic numbers for now or shall we wait for better information?

0b71038 should go into pre13 under any circumstances; I can make it a separate PR if needed.

**Comment by davidlange6** [comment]
@schneiml -lets get an update with documented magic numbers. thx

**Comment by dmitrijus** [comment]
+1

**Comment by davidlange6** [comment]
+1

**Comment by davidlange6** [comment]
+1

**Pull request merged**